src/app/auth/signup/SignUp.tsx

```diff
@@ -0,0 +1,41 @@
+'use client';
+import { useEffect, useState } from 'react';
+import { useFormState } from 'react-dom';
+
+import { signupUser } from '@app/api/user';
+import { AuthError } from '@app/types/list.types';
+
+import AuthLink from '../components/AuthLink';
+import Email from '../components/Email';
+import Password from '../components/Password';
+import SubmitButton from '../components/SubmitButton';
+import Username from '../components/Username';
+const initialState: AuthError = {
+  message: '',
+};
+
+export default function SignUp() {
+  const [state, formAction] = useFormState(signupUser, initialState);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  useEffect(() => {
+    if (state && state.message) {
+      setErrorMessage(state.message);
+    }
+  }, [state]);
+
+  return (
+    <form action={formAction} onSubmit={() => setErrorMessage('')}>
```

Comment:
On submit get 200 Ok, but form says invalid value:
<img width="1464" alt="Untitled" src="https://github.com/user-attachments/assets/eeb2fe51-83e1-47c8-a67a-bea0141ce592">

Also able to create same user more then once

Reply:
with the same username but different emails?

Reply:
I check only emails, I can add to make usernames unique

Reply:
No? I don't think we should make unit username. I meant with same email? Is this works for you? I will check again

Reply:
Yes, I checked to signup with the same email and it shows error. at least in postman

Reply:
I check it again, the promlem was with password, I saw at node part that length has to be at least 7 charecters. And 200 it was about page not an API response.
So it works fine!